crates/update-notifier/src/utils.rs

```diff
@@ -0,0 +1,29 @@
+use std::{
+    path::PathBuf,
+    time::{SystemTime, UNIX_EPOCH},
+};
+
+pub fn ms_since_epoch() -> u128 {
+    SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        .unwrap()
+        .as_millis()
+}
+
+pub fn get_version() -> &'static str {
+    log::debug!("fetching current version");
+    include_str!("../../../version.txt")
```

Comment:
Including outside the crate can break the build if this is ever built from outside this repo, we ran into this with turopack already, would prefer this being somewhat generic 

for the version you can use 
https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-crates

but you'll have to do that in the shim crate (and set it's version accordingly), the env vars are set for each crate